alfresco-zipkin-repo/src/main/java/eu/xenit/alfresco/instrumentation/solr/VirtualSolrSpanFactory.java

```diff
@@ -0,0 +1,221 @@
+package eu.xenit.alfresco.instrumentation.solr;
+
+import brave.Span;
+import brave.Tracer;
+import eu.xenit.alfresco.instrumentation.solr.representations.ShardedSolrRequest;
+import eu.xenit.alfresco.instrumentation.solr.representations.ShardedSolrSubRequest;
+import eu.xenit.alfresco.instrumentation.solr.representations.SolrRequest;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VirtualSolrSpanFactory {
+    private static final Logger logger = LoggerFactory.getLogger(VirtualSolrSpanFactory.class);
+
+    public SolrRequest solrRequest;
+    public boolean sharded = false;
+    public HashMap<String, ShardedSolrRequest> shardedRequests;
+    public String receivingSolrInstance;
+
+    private static final String NUMFOUND_KEY = "numFound";
+    private static final String SUB_NUMFOUND_KEY = "NumFound";
+    private static final String QTIME_KEY = "QTime";
+    private static final String ELAPSEDTIME_KEY = "ElapsedTime";
+    private static final String RESPONSE_KEY = "response";
+    private static final String RESPONSE_HEADER_KEY = "responseHeader";
+    private static final String ORIGINAL_PARAMS_KEY = "_original_parameters_";
+    private static final String SHARDS_KEY = "shards";
+    private static final String DEBUG_KEY = "debug";
+    private static final String TRACK_KEY = "track";
+    private static final String RID_KEY = "rid";
+    private static final String TIMINGS_KEY = "timing";
+    private static final String REQUEST_PURPOSE_KEY = "RequestPurpose";
+    private static final String HIGHLIGHT_QUERY_KEY = "hl.q";
+    private static final String PARSED_QUERY_KEY = "parsedquery_toString";
+
+    public VirtualSolrSpanFactory(JSONObject jsonResponse, String path) {
+        receivingSolrInstance = path;
+
+        //QTime and Number of documents and Query found of main request
+        int numFound = -1;
+        long qTime = -1;
+        String query = "Query not found in response";
+        //Timing information if available
+        JSONObject mainTimings = null;
+
```

Comment:
Should probably have used Jackson here. Problem was that embedded jsons contained in the debug information were not uniformly/consistently named i.e: numFound vs NumFound key

Reply:
If the structure makes it hard to use Jackson to simply have your objects with simple annotations, it's probably not worth the effort to try to use Jackson for this, and just keep your JSONObject approach.

You could do a lot of things with custom deserializers for Jackson, but I think in this case that's more complexity than what you save.